api/segments/models.py

```diff
@@ -198,6 +198,38 @@ def deep_clone(self) -> "Segment":
 
         return cloned_segment
 
+    def match_rules_to_segment(self, segment: "Segment") -> None:
+        _rules = self.rules.all()
+        matched_rules = set()
+        matched_sub_rules = set()
+
+        for rule in segment.rules.all():
+            sub_rules = rule.rules.all()
+            for sub_rule in sub_rules:
+                sub_rule_matched = False
+                for _rule in _rules:
+                    if sub_rule_matched:
```

Comment:
As far as I can tell, the only time we set `sub_rule_matched = True`, we already `break` so do we actually need this, or is it just belt and safeguarding?

Reply:
We break from a more nested part of the codebase. The block that's named 

```python
for _sub_rule in _rule.rules.all():
```

So we need the `sub_rule_matched` bool check to break the parent

```python
for _rule in _rules:
```

Part of the codebase.

Reply:
Ah yes, ok, sorry.